<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>905</width>
    <height>508</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Audio Steganography Decoder</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>0</y>
      <width>597</width>
      <height>455</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="label_title">
       <property name="font">
        <font>
         <family>Century Schoolbook L</family>
         <pointsize>24</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Audio Steganography Decoder</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_type_2">
         <property name="font">
          <font>
           <family>Century Schoolbook L</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Input Files</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_type_4">
           <property name="font">
            <font>
             <family>Century Schoolbook L</family>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Stego File</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="lineEdit_stego">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_decode">
         <property name="text">
          <string>Decode</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_type_3">
         <property name="font">
          <font>
           <family>Century Schoolbook L</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Bits to extract</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_bits_to_extract">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_type">
         <property name="font">
          <font>
           <family>Century Schoolbook L</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Decoding Method</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_LSB">
         <property name="text">
          <string>Least Significant Bit Steganography</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButton_GA">
         <property name="text">
          <string>Genetic Algorithm</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_type_6">
         <property name="font">
          <font>
           <family>Century Schoolbook L</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Message file name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_msgname">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_type_5">
         <property name="font">
          <font>
           <family>Century Schoolbook L</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>LSB number to extract (1-4)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_LSB_nr"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_type_19">
           <property name="font">
            <font>
             <family>Century Schoolbook L</family>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Reading Stego File</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_type_20">
           <property name="font">
            <font>
             <family>Century Schoolbook L</family>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Extracting</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_type_21">
           <property name="font">
            <font>
             <family>Century Schoolbook L</family>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Writing to message file</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QProgressBar" name="progressBar_reading">
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar_extracting">
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar_writing">
         <property name="value">
          <number>24</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>905</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_decode</sender>
   <signal>clicked()</signal>
   <receiver>progressBar_reading</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>477</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_decode</sender>
   <signal>clicked()</signal>
   <receiver>progressBar_extracting</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>485</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_decode</sender>
   <signal>clicked()</signal>
   <receiver>progressBar_writing</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>480</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_LSB</sender>
   <signal>clicked()</signal>
   <receiver>lineEdit_LSB_nr</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>51</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_LSB</sender>
   <signal>clicked()</signal>
   <receiver>label_type_5</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel">
     <x>72</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_GA</sender>
   <signal>clicked()</signal>
   <receiver>label_type_5</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_GA</sender>
   <signal>clicked()</signal>
   <receiver>lineEdit_LSB_nr</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>420</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
